pkg/kubernetes/scripts/policy.js

```diff
@@ -174,6 +177,66 @@
                 return result;
             }
 
+            var sarCache = { };
+
+            function subjectAccessReview(project, user, verb, resource) {
+                var key = (project || "") + ':' + (user ? user.metadata.name : "")+ ':' + verb + ':' + resource;
+                // cached already?
+                if (key in sarCache)
+                    return sarCache[key];
+
+                // pre-fill cache to avoid multiple concurrent requests
+                sarCache[key] = false;
+
+                loader.watch("Projects").then(function() {
+                    // if no project was given, check the first one as an approximation
+                    // TODO: is that sufficient, or should we iterate over all
+                    // projects? (much more expensive)
+                    if (user && !project) {
+                        var projects = select().kind("Project");
+                        for (var p in projects) {
+                            project = projects[p].metadata.name;
+                            break;
+                        }
+                        if (!project)
+                            return false; // the user has no projects
+                    }
+
+                    var request = {
+                        kind: "SubjectAccessReview",
+                        apiVersion: "v1",
+                        namespace: project,
+                        verb: verb,
+                        resource: resource
+                    };
+
+                    methods.post(loader.resolve("subjectaccessreviews"), request)
+                        .then(function(response) {
+                            sarCache[key] = response.allowed;
+                            // tell UI to update only if allowed, as our "no lookup yet" default is false already
+                            if (response.allowed) {
+                                $rootScope.$applyAsync();
```

Comment:
Is your concern about using `applyAsync()` at all? We somehow need to tell the dashboard to reevaluate the `ng-if=` condition (I tested this with wrapping this into an additional 10-second delay) once the result comes in. I modelled this after the existing `fillWhoCan()` and `expireWhoCan()` which structurally do a very similar thing and also use `applyAsync()` in the end.

Or is it just that *this* particular function should not do it, and there should be some indirection? I. e. `subjectAccessReview()` just sets a property and dashboard watches for changes of that (I assume that's possible in JS) and calls applyAsync()?